### PR TITLE
New Gun Sounds

### PR DESCRIPTION
**Pull request by granodd**
Adds a whole lot of sounds to ballistic-based guns. Including fire sounds, loading and interaction sounds, and sounds for casings, metal and plastic, hitting the ground. Moved all of the ballistic based sounds to their own folders within the **sound/weapons** folder under **sound/weapons/gunshots** and **sound/weapons/gun_interactions**.

This PR also refactors reloading for projectile and automatic weapons, as well as the SAW.

Credit to Baystation for the gun firing and reloading sounds, as well as TG for its shotgun loading and bullet insertion sounds, and to Gamma station for the sound used for the laser carbine and heavy pistols.

Minor things: added comment labels to some weapons in their respective folders to facilitate finding specific guns.

**Changelog:**
:cl: name here
**soundadd:** Sounds for pistols, heavier pistols (Deagle), revolvers, shotguns, snipers, rifles, heavy machine guns, and the laser carbine.
**soundadd:** Sounds for loading magazines into weapons for pistols, heavier pistols, SMGs, assault rifles, loading bullets into shotguns, loading bullets into revolvers, and loading bullets into magazines.
**soundadd:** Gun interaction sounds for swapping fire mode, ejecting the chambered bullet, and rearming mechs
**tweak:** Refactored reloading for projectile and automatic weapons to be less awful (thanks to @farie82 for assistance in this)
**tweak:** Moved ballistic gun sounds to their own folder.
/:cl:

### CODE REVIEW

**Comment by variableundefined** [comment]
A requested PR. Need to review it though.

File: code/modules/projectiles/guns/projectile.dm [hunk 1]
```diff
@@ -6,8 +6,12 @@
 	w_class = WEIGHT_CLASS_NORMAL
 	materials = list(MAT_METAL=1000)
 
+	light_power = 10
+	light_color = LIGHT_COLOR_WHITE //gun flash variables
```
Comment (Crazylemon64) [inline code/modules/projectiles/guns/projectile.dm]: What's going on here?

Comment (granodd) [reply code/modules/projectiles/guns/projectile.dm]: Oh woops, code I forgot to take out. This was part of me attempting to put in weapon flashes before I eventually decided to do that in its own PR.

File: code/modules/projectiles/guns/projectile.dm [hunk 2]
```diff
@@ -51,21 +52,34 @@
...
-			update_icon()
-			return 1
-		else if(magazine)
-			to_chat(user, "<span class='notice'>There's already a magazine in \the [src].</span>")
+		if(istype(AM, mag_type))
+			if(can_reload())
+				reload(AM, user)
+				to_chat(user, "<span class='notice'>You load a new magazine into \the [src].</span>")
+				return TRUE
+			else if(!can_reload() && !can_tactical)
```
Comment (farie82) [inline code/modules/projectiles/guns/projectile.dm]: no need to check !can_reload() here again. In the if statement it has concluded that the value was false so this one is always !(FALSE) which is just TRUE

File: code/modules/projectiles/guns/projectile.dm [hunk 3]
```diff
@@ -51,21 +52,34 @@
...
+				to_chat(user, "<span class='notice'>There's already a magazine in \the [src].</span>")
+				return TRUE
```
Comment (farie82) [inline code/modules/projectiles/guns/projectile.dm]: Making an else here to tactically reload might be a good idea. Due to you including can_tactical in the projectile class

File: code/modules/projectiles/guns/projectile/automatic.dm [hunk 4]
```diff
@@ -26,20 +27,13 @@
...
-			user.remove_from_mob(AM)
-			magazine = AM
-			magazine.loc = src
-			chamber_round()
-			A.update_icon()
-			update_icon()
-			return 1
+				magazine = null		
+			reload(AM, user)			
+			return
```
Comment (farie82) [inline code/modules/projectiles/guns/projectile/automatic.dm]: This part can be moved to the attackby from projectile due to the reason stated before.

**Comment by granodd** [comment]
Moved all of the reload code (disregarding the SAW's) to the projectiles/guns/projectile.dm.

**Comment by Fox-McCloud** [comment]
I'd really be better to isolate this to just a sound change. 

I'd recommend slotting in TG's gun/projectile sound system for parity reasons; we've kept in line with their guns so far, and it'll make updating+dealing with the system far easier, for everyone involved.

**Comment by Shadeykins** [comment]
This may be a big ask, but it may be worthwhile to create a short .webm that showcases the new sounds live.

**Comment by KasparoVy** [comment]
Reminds me of ye olde https://www.youtube.com/watch?v=4_swc5ko_24

**Comment by granodd** [comment]
Removed the casing drop sound from removing a chambered round in a gun, as it was playing the metal casing sound for shotgun rounds. Rather than find some messy way to fix this I've opted to leave it out. A showcase of the weapon sounds can be found here by request of @Shadeykins : 
https://youtu.be/6Tg9UvoImYw

@Fox-McCloud What specifically are you referring to when you say "gun/projectile" sound system from TG?

**Comment by KasparoVy** [comment]
Couldn't you have the casing sound be a property associated with the magazine/projectile and then pull from that in the instance above? That'd let you define custom ones for, say, shotgun shells.

**Comment by Aurorablade** [comment]
Fox may mean this https://github.com/tgstation/tgstation/pull/41577
https://github.com/tgstation/tgstation/pull/41620

Might be wrong anyway kind share what he said there.

**Comment by variableundefined** [comment]
@granodd I apologize for the late reply. I've read through the PRs and relevant comments.

I believe when Fox referred to the tg parity system he meant the lack of magin and magout sounds. 

I've looked at the PRs Fethas linked and it seems the TG maintainers deem it OK to have separate dry fire and suppressed sounds so I am personally OK with us having magin and magout sounds. Should resource file bloat turn out to be an issue later on we could consider consolidating these sounds.

So with that being said, I support this PR and I would like to merge it once the following two concerns are addressed:
1. The PR is deconflicted
2. All sounds resources name are changed to full lowercase and the references to them at also changed to full lowercase. Inconsistent casing is one of the cause of linux issue and had also led to resource file corruption. We are now enforcing it as a standard.

**Comment by granodd** [comment]
Been a while since I've fiddled with any of this. Changed file names to be lower case as per request as well as resolving some conflicts (hopefully).

@Nicaragua1 listening to it, it might be. It came from Bay.

**Comment by KasparoVy** [comment]
Fuck yes, nicely done

**Pull request merged**